test/src/org/labkey/test/components/targetedms/UtilizationCalendarWebPart.java

```diff
@@ -73,7 +73,7 @@ public UtilizationCalendarWebPart markOffline(String startDate, @Nullable String
         ModalDialog modalDialog = clickDate(startDate);
         getWrapper().setFormElement(Locator.name("event-description").findElement(modalDialog.getComponentElement()), description);
         getWrapper().setFormElement(Locator.name("event-end-date").findElement(modalDialog.getComponentElement()), endDate);
-        modalDialog.dismiss("Save", 0);
+        doAndWaitForElementToRefresh(() -> modalDialog.dismiss("Save", 0), elementCache().dayLoc.withText(extractDay(startDate)), 10);
```

Comment:
FYI, none of the calls to `modalDialog.dismiss` in this class actually wait for the dialog to go away. Making them do so might improve its general reliability.

```suggestion
        doAndWaitForElementToRefresh(() -> modalDialog.dismiss("Save"), elementCache().dayLoc.withText(extractDay(startDate)), 10);
```